protocols/gossipsub/src/behaviour/tests.rs

```diff
@@ -679,17 +682,21 @@ fn test_join() {
 /// Test local node publish to subscribed topic
 #[test]
 fn test_publish_without_flood_publishing() {
```

Comment:
can we not change the existing tests and instead add specific tests for the new configurations per topic?